### PR TITLE
Speed desaturation

### PR DESCRIPTION
**Pull request by Kenneth-Olibrice**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/SwerveDrive.java [hunk 1]
```diff
@@ -188,12 +188,16 @@ public void move(ChassisSpeeds chassisSpeeds) {
       return;
     }
     SwerveModuleState[] states = m_kinematics.toSwerveModuleStates(chassisSpeeds);
+    ChassisSpeeds currentSpeed = m_kinematics.toChassisSpeeds(getModuleStates()); // TODO TEST THIS
```
Comment (averwhy) [inline src/main/java/frc/robot/subsystems/SwerveDrive.java]: you have a TODO in your code

**Review by averwhy — DISMISSED** [review]

**Review by averwhy — COMMENTED** [review]
much better

**Review by averwhy — COMMENTED** [review]
not working atm afaik, made draft until we test again 👍

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -257,11 +257,13 @@ private void testCommands() {
...
-    m_tester.a().whileTrue(new DriveToTarget(m_swerveDrive, new Pose2d(-1, 0, Rotation2d.fromDegrees(0)), Constants.DriveToTargetTolerance).repeatedly());
-    m_tester.x().whileTrue(new DriveToTarget(m_swerveDrive, new Pose2d(0, 1, Rotation2d.fromDegrees(0)), Constants.DriveToTargetTolerance).repeatedly());
+    var xStart = 8;
+    var yStart = 4;
+    m_tester.start().onTrue(new ResetPose(m_swerveDrive, new Pose2d(xStart, yStart, Rotation2d.fromDegrees(0))));
+    m_tester.b().whileTrue(new DriveToTarget(m_swerveDrive, new Pose2d(xStart, yStart, Rotation2d.fromDegrees(0)), Constants.DriveToTargetTolerance));
+    m_tester.y().whileTrue(new DriveToTarget(m_swerveDrive, new Pose2d(xStart + 1, yStart, Rotation2d.fromDegrees(90)), Constants.DriveToTargetTolerance));
+    m_tester.a().whileTrue(new DriveToTarget(m_swerveDrive, new Pose2d(xStart - 1, yStart, Rotation2d.fromDegrees(270)), Constants.DriveToTargetTolerance));
+    m_tester.x().whileTrue(new DriveToTarget(m_swerveDrive, new Pose2d(xStart, yStart + 1, Rotation2d.fromDegrees(180)), Constants.DriveToTargetTolerance));
```
Comment (jnallard) [inline src/main/java/frc/robot/RobotContainer.java]: Added this to this branch as we will test this with the desaturation changes tomorrow

**Review by jnallard — APPROVED** [review]

**Review by tri-vu-12 — APPROVED** [review]
Looked over as a team!

**Pull request merged**